services/ui-src/src/components/reports/EntityOverlayPage.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from "react";
+// components
+import { Box, Button, Image, useDisclosure } from "@chakra-ui/react";
+import {
+  Alert,
+  Form,
+  ReportPageFooter,
+  ReportPageIntro,
+  CloseEntityModal,
+} from "components";
+// types
+import { EntityShape, AlertTypes, EntityOverlayPageShape } from "types";
+// assets
+import closeIcon from "assets/icons/icon_close.png";
+
+export const EntityOverlayPage = ({ route, validateOnRender }: Props) => {
+  const submitting = false;
+  const [selectedEntity, setSelectedEntity] = useState<EntityShape | undefined>(
```

Comment:
I wonder if in the future we'll grab this from the Zustand state? 🤔 Might be a better way to grab but this can work for now.

Reply:
you're right! i believe i technically do not need this code for the UI, and it becomes more important for future work in getting the logic up and running. i got rid of it for now + will chat with you about how to correctly grab from Zustand